### DISCUSSION TITLE
Problem with download m4a

### DISCUSSION THREAD

**Discussion by LuluDH** [aclap-dev/video-downloadhelper#350] [Help and Q&A]
### Your question

Hello. When I used video dowloadhelper, to download a m4a audio, the download is stopped. For example :

SideDownload error:

[mp3 @ 000001e264b5ddc0] Invalid audio stream. Exactly one MP3 audio stream is required.

[out#0/mp3 @ 000001e264b4c180] Could not write header (incorrect codec parameters ?): Invalid argument

Error opening output file C:\Users\StarÉcolO\Desktop\Simon Gronowski  Ma vie est un miracle.mp3.

Error opening output files: Invalid argument

Thanks a lot for your answer.
(I am French, excuse me for "my english")

### Addon & CoApp version and details

version: 8.2.2.8
channel: stable
build date: 2024-03-18
build options: linuxlic: false, noyt: false, browser: firefox
lang: fr
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:124.0) Gecko/20100101 Firefox/124.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\StarÉcolO",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by jcfrog** [comment]
Hi,
Can you share a link of a public video with this issue?

**Reply by LuluDH** [reply]
Hello,
It's for example, this audio :
https://www.radiofrance.fr/franceinter/podcasts/les-odyssees/simon-gronowsky-1560333
Usually, I click on "écouter" and the audio begin, and I download the audio with download helper.
But this audio is m4a and download helper jams and fails.
It's the same with : 
https://www.radiofrance.fr/franceinter/podcasts/les-odyssees/vincent-munier-sur-les-traces-du-loup-des-glaces-7941438
https://www.radiofrance.fr/franceinter/podcasts/les-odyssees/oum-kalsoum-la-diva-du-moyen-orient-5217957

Thanks a lot for your help.

**Reply by Mattmjs** [reply]
I see the problem. VDH is trying to download the m4a file which uses the aac codec as mp3. If it is downloaded as m4a the
download works fine without any issue.

**Reply by LuluDH** [reply]
Thanks for this idea. How choose to download as m4a in the settings / codec of video download helper ?
Thanks,

**Reply by Mattmjs** [reply]
I don't think there is a way to do that, the extension might need an update for m4a files. You can save these files yourself.
Just go into the hit details for the file and look for the link shown in the image. Open this link in a new tab.
The audio will be playing, to save just right click on the player and click save audio as.

![radiofrance](https://github.com/aclap-dev/video-downloadhelper/assets/160297904/ff244a9f-3b17-486d-acd3-1df64453e87a)

**Reply by LuluDH** [reply]
Thanks a lot ! I have succeeded !
